core/src/main/scala/org/zalando/grafter/Rewriter.scala

```diff
@@ -173,6 +175,22 @@ trait Rewriter {
 
     results.toList
   }
+
+  def topBreadthfirst(s: Strategy): Strategy = {
+    def bf(s1: =>Strategy): Strategy =
+      one(s1) <+ one(bf(s1))
+
+    s <+ bf(s)
+  }
+
+  private def canBeRewritten(v: Any): Boolean = v match {
```

Comment:
This code is similar to the code in `Rewriter.all` which defines which children of a node can be rewritten.